tests/escape.temper

```diff
@@ -30,6 +30,11 @@ test("escaped square brace") {
   }
 }
 
+test("escape dash in square") {
+  let regex = compile(raw"[\-]");
```

Comment:
You can already put dash at the front of `[]` without escape, but say `[a-]` would fail, so it's nice to allow escaping it.

Reply:
Or maybe `]` should end either way like I think might be common out there, but escape is still nice for arbitrary location.